lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1158,7 +1158,11 @@ void addIntelFPGADecorations(
   for (const auto &I : Decorations) {
     if (I.first == DecorationMemoryINTEL)
       E->addDecorate(new SPIRVDecorateMemoryINTELAttr(E, I.second));
-    else {
+    else if (I.first == DecorationRegisterINTEL ||
+             I.first == DecorationSinglepumpINTEL ||
+             I.first == DecorationDoublepumpINTEL) {
+      E->addDecorate(I.first);
```

Comment:
You can also add an assertion that `I.second` is an empty `StringRef`

Reply:
Done